admin-manual/security/csp-headers.rst

```diff
@@ -0,0 +1,263 @@
+       participant Server
+
+       Browser ->> Server: HTTP Request
+       note over Server: Generate unique nonce and<br/>store in session variable
+       note over Server: Build CSP header directive including nonce to use in reponse
+
+       note over Server: Tag each inline script with nonce
+       Server ->> Browser: HTTP Response
+       note over Browser: Verify reponse sources<br/>against values in CSP directive<br />rejecting any that do not match
+
+
+.. _csp-configuration:
+
+AtoM's CSP configuration
+------------------------
+
+This is the default configuration you'll find in AtoM's :ref:`config-app-yml` file:
+
+.. code-block:: yaml
+
+   # Content Security Policy (CSP) header configuration. CSP settings apply
+   # only when a B5 theme is active, else these settings are bypassed.
+   csp:
+       response_header: Content-Security-Policy
+       directives: "default-src 'self'; font-src 'self'; img-src 'self' https://www.gravatar.com/avatar/ blob:; script-src 'self' 'nonce'; style-src 'self' 'nonce'; worker-src 'self' blob:; frame-ancestors 'self';"
+
+The CSP ``repsonse_header`` setting is used to set the CSP header type and can have one of
+two values:
+
+- **Content-Security-Policy**: 
+  This setting will enforce the defined policy. When set, the browser will block any
+  resources (scripts, images, stylesheets, etc.) that violate the policy directives.
+
+- **Content-Security-Policy-Report-Only**:
+  This header works similarly to the ``Content-Security-Policy`` but does not block any
+  resources allowing developers to test and monitor potential violations without
+  affecting the functionality of the web page. This is useful for testing a new policy or
+  changes to an existing policy without risking breakage. Violations will be reported to
+  the browser's console.
+
+The CSP ``directives`` setting contains the CSP policy that will be sent in the CSP header.
+The value for the ``directives`` setting above is the default when the Dominion theme is
+in use. Multiple directives get delineated with a semicolon (``;``).
+
+.. IMPORTANT::
+    The value ``nonce`` in the above example is a placeholder and will be replaced
+    with a generated unique value when AtoM generates the page response.
+
+The specific directives defined in the defaults above are as follows:
+
+- default-src: if resources aren't mentioned in other sections, this policy is applied.
+- font-src: stipulates which fonts can be loaded.
+- img-src: stipulates which images can be loaded.
+- script-src: stipulates which scripts can be loaded.
+- style-src: stipulates which styles or CSS can be loaded.
+- frame-ancestors 'self': determines which domains can embed the page as a frame.
+
+Specifying ``self`` ensures that only trusted resources from the *same origin* are loaded
+and executed.
+
+Updating these settings will require restarting **php-fpm**.
+
+CSP headers will only be applied to responses if a Bootstrap 5 based theme is in use.
```

Comment:
I'd suggest putting this in an IMPORTANT admonition at the bottom of this section, since a) we want to emphasize it now, but b) in 2.9 and later we can remove that mention, since there will only be B5 themes. i.e.

```
.. IMPORTANT::

   CSP headers will only be applied toa response if a Bootstrap 5 based theme is in use. See: 

   * :ref:`customization-theming`
   * :ref:`themes`
```

Reply:
Good idea! Added...